File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # bad solution quadratic complexity
        # set_nums = set()
        # for i in nums: 
        #     if i in set_nums:
        #         return True
        #     else: 
        #         set_nums.add(i)
        # return False
        # Linear Complexity
        if len(set(nums)) != len(nums):
            return True
        return False

File: 0217-contains-duplicate/0217-contains-duplicate.rs
use std::collections::HashSet;
impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        // quadratic Complexity
        // for (ind_first, i) in nums.iter().enumerate(){
        //     for (ind, j) in nums.iter().enumerate(){
        //         if i == j && ind_first != ind{
        //             return true
        //         }
        //     }
        // }

        // return false

        // Linear Complexity
        let x: HashSet<_> = nums.iter().collect();
        if x.len() == nums.len(){
            return false;
        }
        return true;   
    }
}